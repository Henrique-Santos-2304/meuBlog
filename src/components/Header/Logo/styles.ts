import styled, { css } from "styled-components";
export const Wrapper = styled.div`
  padding: 0.3rem 2rem;
  background: transparent;
`;

export const PreText = styled.p`
  ${({ theme }) => css`
    font-family: ${theme.font.family.Tangerine};
    font-size: 4.5rem;
    color: ${theme.colors.white};
    span {
      color: ${theme.colors.red800};
    }
  `}
`;
