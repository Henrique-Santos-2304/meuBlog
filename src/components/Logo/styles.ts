import styled, { css } from "styled-components";
import media from "styled-media-query";

export const Wrapper = styled.div`
  ${({ theme }) => css`
    width: max-content;
    display: flex;
    justify-content: center;
    align-items: center;
    padding: ${theme.spacings.xxsmall} ${theme.spacings.xsmall};
    background: ${theme.colors.mainBg};
    border-radius: ${theme.border.radius};
  `}
`;
export const PreTextLogo = styled.p`
  ${({ theme }) => css`
    font-family: ${theme.font.family.Tangerine};
    font-size: 5rem;
    color: ${theme.colors.white};
  `}
  ${media.lessThan("small")`
    font-size: 3rem;
  `}
`;

export const ProTextLogo = styled.p`
  ${({ theme }) => css`
    font-family: ${theme.font.family.Tangerine};
    font-size: 5rem;
    color: ${theme.colors.red800};
  `}
  ${media.lessThan("small")`
    font-size: 3rem;
  `}
`;
