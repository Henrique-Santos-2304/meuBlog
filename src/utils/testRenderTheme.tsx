import { ThemeProvider } from "styled-components";
import { ChakraProvider } from "@chakra-ui/react";

import { render, RenderResult } from "@testing-library/react";
import theme from "styles/theme";
export const renderTheme = (children: React.ReactNode): RenderResult => {
  return render(
    <ThemeProvider theme={theme}>
      <ChakraProvider> {children}</ChakraProvider>
    </ThemeProvider>
  );
};
