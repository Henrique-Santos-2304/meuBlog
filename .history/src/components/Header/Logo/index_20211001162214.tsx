import * as S from "./styles";
import Henrique from "../../../public/img/hero-illustration.svg";
import Image from "next/image";

const Logo = () => (
  <S.Wrapper>
    <S.Image>
      <Image src={Henrique} height="100%" width="100%"></Image>
    </S.Image>
    <S.TextWrapper>
      <S.Tittle>
        Tech <S.Title2>Web</S.Title2>
      </S.Tittle>
      <S.Name> Henrique dos Santos </S.Name>
    </S.TextWrapper>
  </S.Wrapper>
);

export default Logo;
