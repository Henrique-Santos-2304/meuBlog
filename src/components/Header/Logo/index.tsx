import * as S from "./styles";
import { Box } from "@chakra-ui/react";

const Logo = () => (
  <S.Wrapper>
    <Box padding="4" maxW="3xl">
      <S.PreText aria-label="logo dev">
        Web<span>Dev</span>
      </S.PreText>
    </Box>
  </S.Wrapper>
);

export default Logo;
