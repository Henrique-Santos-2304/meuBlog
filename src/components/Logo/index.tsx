import * as S from "./styles";

const Logo = () => (
  <S.Wrapper>
    <S.PreTextLogo>Web</S.PreTextLogo>
    <S.ProTextLogo>Dev</S.ProTextLogo>
  </S.Wrapper>
);

export default Logo;
