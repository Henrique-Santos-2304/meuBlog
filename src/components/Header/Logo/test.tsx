import Logo from ".";
import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";

describe("<Logo />", () => {
  it("should render ", () => {
    const { container } = renderTheme(<Logo />);
    expect(container).toMatchSnapshot();
  });
  it("should logo text in the display ", () => {
    renderTheme(<Logo />);
    expect(screen.getByLabelText(/logo dev/i)).toBeInTheDocument();
  });
  it("should have a text Web Dev  ", () => {
    renderTheme(<Logo />);
    expect(screen.getByText(/web/i)).toBeInTheDocument();
    expect(screen.getByText(/dev/i)).toBeInTheDocument();
  });
});
