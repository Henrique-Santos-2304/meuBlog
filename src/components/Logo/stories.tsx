import { Story, Meta } from "@storybook/react";
import Logo from ".";

export default {
  title: "Logo",
  component: Logo,
} as Meta;

export const Basic: Story = (args) => <Logo {...args} />;

Basic.args = {
  title: "title basic",
  description: "description basic",
};

export const Default: Story = (args) => <Logo {...args} />;
