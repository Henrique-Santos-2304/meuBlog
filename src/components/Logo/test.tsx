import { screen } from "@testing-library/react";
import { renderTheme } from "utils/testRenderTheme";
import theme from "styles/theme";
import "jest-styled-components";

import Logo from ".";

describe("<Logo />", () => {
  it("create Snapshot for element", () => {
    const { container } = renderTheme(<Logo />);
    expect(container).toMatchSnapshot();
  });
  it("should have a text Web", () => {
    renderTheme(<Logo />);
    expect(screen.getByText(/Web/i)).toBeInTheDocument();
    expect(screen.getByText(/Web/i)).toHaveStyle({ color: theme.colors.white });
  });
  it("should have a text Dev", () => {
    renderTheme(<Logo />);
    expect(screen.getByText(/Dev/i)).toBeInTheDocument();
    expect(screen.getByText(/Dev/i)).toHaveStyle({
      color: theme.colors.red800,
    });
  });
});
