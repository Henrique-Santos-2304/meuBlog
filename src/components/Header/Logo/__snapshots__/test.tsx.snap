// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render  1`] = `
.c0 {
  padding: 0.3rem 2rem;
  background: transparent;
}

.c1 {
  font-family: 'Tangerine',cursive;
  font-size: 4.5rem;
  color: #E5E7EB;
}

.c1 span {
  color: #C11010;
}

<div>
   
  <div
    class="c0"
  >
    <div
      class="css-hl966"
    >
      <p
        aria-label="logo dev"
        class="c1"
      >
        Web
        <span>
          Dev
        </span>
      </p>
    </div>
  </div>
</div>
`;
